src/std/text/json/input.ss

```diff
@@ -45,29 +47,34 @@
 
            (def (read-json-hash input env)
              (read-char input)
-             (let (obj (if (&env-symbolic-keys env)
-                         (make-hash-table-eq)
-                         (make-hash-table)))
+             (let ((obj (if (&env-symbolic-keys env)
+                          (make-hash-table-eq)
+                          (make-hash-table)))
+                   (lst '()))
                (let lp ()
                  (let (key (read-json-hash-key input env))
-                   (if key
+                   (when key
```

Comment:
that looks wrong, you'll get a void if the test is false.

Reply:
I change the parenthesization so the lp is not in tail position. What comes after is, that handles returning the right kind of hash-table or walist.